### PR TITLE
fix: Fix credentials interceptions & autoLogin/logout

### PR DESCRIPTION
**Pull request by KillianCourvoisier**
This PR fixes the credentials interception and usage.
It also ensure a thinner logout verifications.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -146,23 +146,25 @@ class SoshContentScript extends ContentScript {
...
     await this.navigateToLoginForm()
-
-    if (credentials) {
-      this.log('info', 'found credentials, processing')
-      await this.tryAutoLogin(credentials)
-    } else {
-      this.log('info', 'no credentials found, use normal user login')
-      await this.waitForUserAuthentication()
+    if (
+      (await this.isElementInWorker('#password')) ||
```
Comment (doubleface) [inline src/index.js:156]: FYI: you can replace both isElementInWorker with :

```javascript
await this.isElementInWorker('#password, #login')
```

File: src/index.js [hunk 2]
```diff
@@ -275,9 +277,17 @@ class SoshContentScript extends ContentScript {
     const emailSelector = '#login'
     const passwordInputSelector = '#password'
     const loginButtonSelector = '#btnSubmit'
-    await this.waitForElementInWorker(emailSelector)
-    await this.runInWorker('fillForm', credentials)
-    await this.runInWorker('click', loginButtonSelector)
+    await this.PromiseRaceWithError(
+      [
+        this.waitForElementInWorker(emailSelector),
```
Comment (doubleface) [inline src/index.js:282]: Same here and then you don't need the race :

```javascript
await this.waitForElementInWorker(`${emailSelector}, ${passwordInputSelector}`)
```

File: src/index.js [hunk 3]
```diff
@@ -312,7 +315,14 @@ class SoshContentScript extends ContentScript {
     if (this.store.userCredentials != undefined) {
       await this.saveCredentials(this.store.userCredentials)
     }
-
+    if (await this.isElementInWorker('#password')) {
+      await this.evaluateInWorker(function reload() {
```
Comment (doubleface) [inline src/index.js:319]: why this reload ?

Comment (KillianCourvoisier) [reply src/index.js:319]: Because on some execution, the login page load but stay on the form, reloading seems to set this up correctly

**Review by doubleface — APPROVED** [review]

**Pull request merged**